File: DSA/Manav-Rajput/Day59_question1.java
class Solution {
    Node compute(Node head) {
        if (head == null || head.next == null) {
            return head;
        }
        
        // Reverse the linked list
        head = reverseList(head);
        
        // Traverse the reversed list and keep track of the max value
        Node maxNode = head;
        Node current = head;
        
        while (current != null && current.next != null) {
            if (current.next.data < maxNode.data) {
                // Remove the node
                current.next = current.next.next;
            } else {
                // Move to the next node
                current = current.next;
                maxNode = current;
            }
        }
        
        // Reverse the list again to restore the original order
        head = reverseList(head);
        
        return head;
    }
    
    // Helper method to reverse the linked list
    Node reverseList(Node head) {
        Node prev = null;
        Node current = head;
        Node next = null;
        
        while (current != null) {
            next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }
        
        return prev;
    }
}

File: DSA/Arshdeep-Kaur/Day 59/Day59_Question1.cpp
class Solution {
  public:
    Node *compute(Node *head) {
        if (head == NULL || head->next == NULL) return head;

        // Reverse the linked list
        Node* prev = NULL;
        Node* curr = head;
        Node* next = NULL;
        
        while (curr != NULL) {
            next = curr->next;
            curr->next = prev;
            prev = curr;
            curr = next;
        }
        
        head = prev;
        Node* maxNode = head;
        curr = head->next;
        prev = head;
        
        // Traverse the reversed list and remove nodes
        while (curr != NULL) {
            if (curr->data < maxNode->data) {
                prev->next = curr->next;
            } else {
                maxNode = curr;
                prev = curr;
            }
            curr = curr->next;
        }
        
        // Reverse the list again to restore the original order
        prev = NULL;
        curr = head;
        next = NULL;
        
        while (curr != NULL) {
            next = curr->next;
            curr->next = prev;
            prev = curr;
            curr = next;
        }
        
        return prev; // The new head of the modified list
    }
};
